core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/async/SimpleHttpContentPublisher.java

```diff
@@ -62,12 +62,14 @@ private SubscriptionImpl(Subscriber<? super ByteBuffer> s) {
 
         @Override
         public void request(long n) {
-            if (n <= 0) {
-                s.onError(new IllegalArgumentException("Demand must be positive"));
-            } else if (running) {
-                s.onNext(ByteBuffer.wrap(content));
-                s.onComplete();
+            if (running) {
```

Comment:
Do we need to use an atomic boolean here?

Reply:
I don't think we need to

> A Subscriber MUST ensure that all calls on its Subscription take place from the same thread or provide for respective external synchronization.